pkg/ingester/stream_test.go

```diff
@@ -334,21 +338,24 @@ func TestPushRateLimit(t *testing.T) {
 		{Timestamp: time.Unix(1, 0), Line: "aaaaaaaaab"},
 	}
 	// Counter should be 2 now since the first line will be deduped.
-	_, err = s.Push(context.Background(), entries, recordPool.GetRecord(), 0, true)
+	_, err = s.Push(context.Background(), entries, recordPool.GetRecord(), 0, true, true)
+	require.Error(t, err)
 	require.Contains(t, err.Error(), (&validation.ErrStreamRateLimit{RateLimit: l.PerStreamRateLimit, Labels: s.labelsString, Bytes: flagext.ByteSize(len(entries[1].Line))}).Error())
 }
 
 func TestPushRateLimitAllOrNothing(t *testing.T) {
 	l := validation.Limits{
 		PerStreamRateLimit:      10,
 		PerStreamRateLimitBurst: 10,
+		ShardStreams: &shardstreams.Config{
```

Comment:
I don't think we need this setup because the limits for this are used in `instance.go` to determine `rateLimitWholeStream`

We should probably have a test in `instance.go` to make sure this stays wired together the same way in the future.

Reply:
Good call. I added a test covering it, WDYT?